src/__tests__/VoucherRedemption.tsx

```diff
@@ -0,0 +1,263 @@
+  });
+
+  test('should display VoucherPage Amount prompt when clicking "Click to being to being reddeming your voucher"', async () => {
+    const { findByText, findAllByText } = renderIntegration(
+      '/voucher/123asdf',
+      renderVoucher(providerValues)
+    );
+
+    const NextButton = await findByText(
+      'Click to begin redeeming your voucher'
+    );
+
+    fireEvent.click(NextButton);
+
+    const Header = await findByText(`Shunfa Bakery`, { exact: false });
+    const CurrentBalanceHeader = await findByText('Current balance');
+    const Amount = await findAllByText('25.00', { exact: false });
+    const Prompt = await findByText('How much are you spending today?');
+    const Code = await findByText('123asdf', { exact: false });
+    const CodeLabel = await findByText('Voucher Code', { exact: false });
+    const Next = await findByText('Next', { exact: false });
+
+    expect(CurrentBalanceHeader).toBeInTheDocument();
+    expect(Header).toBeInTheDocument();
+    expect(Prompt).toBeInTheDocument();
+    expect(Amount.length).toBe(2);
+    expect(CodeLabel).toBeInTheDocument();
+    expect(Code).toBeInTheDocument();
+    expect(Next).toBeInTheDocument();
+  });
+
+  test('should display VoucherPage Confirm prompt when clicking "Next"', async () => {
+    const { findByText, findByPlaceholderText } = renderIntegration(
+      '/voucher/123asdf',
+      renderVoucher(providerValues)
+    );
+
+    const NextButton = await findByText(
+      'Click to begin redeeming your voucher'
+    );
+    fireEvent.click(NextButton);
+
+    const Input = await findByPlaceholderText(`0.00`, { exact: false });
+    const Next = await findByText('Next', { exact: false });
+    fireEvent.change(Input, { target: { value: 22 } });
+    fireEvent.click(Next);
+
+    const Header = await findByText(`Shunfa Bakery`, { exact: false });
+    const PageHeader = await findByText('Complete Your Purchase');
+    const VoucherBalance = await findByText('25.00', { exact: false });
+    const RemainingBalance = await findByText('3.00', { exact: false });
+    const RedemptionAmount = await findByText('22.00', { exact: false });
+    const Code = await findByText('123asdf', { exact: false });
+    const CodeLabel = await findByText('Voucher Code', { exact: false });
+    const ButtonLabel = await findByText(
+      'Please show your phone to the merchant cashier to confirm the purchase.',
+      { exact: false }
+    );
+    const Next1 = await findByText('Next', { exact: false });
+
+    expect(PageHeader).toBeInTheDocument();
+    expect(Header).toBeInTheDocument();
+    expect(VoucherBalance).toBeInTheDocument();
+    expect(RemainingBalance).toBeInTheDocument();
+    expect(RedemptionAmount).toBeInTheDocument();
+    expect(CodeLabel).toBeInTheDocument();
+    expect(ButtonLabel).toBeInTheDocument();
+    expect(Code).toBeInTheDocument();
+    expect(Next1).toBeInTheDocument();
+  });
+
+  test('should display VoucherPage Completed prompt', async () => {
```

Comment:
Is it possible to nest these tests and let them test first next, second next, third next, completed?

Reply:
So i had to think about this alot. In general, nesting tests are a little tricky because of the use of reassignment of variables, and the clean up is pretty messy so i try to avoid it at all times. For this case we could nest the tests because we dont do anything too crazy like variable reassignment. If anything, this flow showed the flaws of my design in handling the voucher redemption page, but thats something to consider later on, when we actually get more feedback on the handling of the voucher redemption flow.